internal/bgp/messages.go

```diff
@@ -297,11 +304,18 @@ func sendUpdate(w io.Writer, asn uint32, ibgp bool, defaultNextHop net.IP, adv *
 		return err
 	}
 	l := b.Len()
+	legacyAnnouncements := []*net.IPNet{adv.Prefix}
+	if adv.Prefix.IP == nil {
+	    return nil
+	}
+	if len(adv.Prefix.IP) == net.IPv6len {
```

Comment:
You need to do a bit more work here unfortunately, because IPv4 addresses can be encoded into 16 bytes in Go. So, you can't know if the IP is IPv4 or IPv6 just based on length.

Instead, you should do:

```go
isIPv6 := adv.Prefix.IP.To4() == nil
```

This tries to convert the IP to the canonical 4-byte IPv4 encoding, and returns nil if that's not possible (i.e. the address is IPv6).

Then you can use `if (isIPv6) ...` in checks.

Reply:
This should now be fixed, thank you!